llvm/test/CodeGen/AMDGPU/build-vector-packed-partial-undef.ll

```diff
@@ -244,7 +244,8 @@ define void @undef_hi_op_v2f16(half %arg0) {
 ; GFX9-LABEL: undef_hi_op_v2f16:
 ; GFX9:       ; %bb.0:
 ; GFX9-NEXT:    s_waitcnt vmcnt(0) expcnt(0) lgkmcnt(0)
-; GFX9-NEXT:    v_pk_add_f16 v0, v0, 1.0 op_sel_hi:[1,0]
+; GFX9-NEXT:    v_add_f16_e32 v0, 1.0, v0
+; GFX9-NEXT:    v_pack_b32_f16 v0, v0, v0
```

Comment:
Yeah, I've marked this patch as a draft since I just want to share my findings. Hoping others might have some ideas on additional patterns that can be matched to mitigate all these regressions